### PR TITLE
Speed up smart indenting in the presence of many inactive regions

### LINKED ISSUE #5394: Perf issues editing files with nested #if and #region
1. Create a C# Console App with the code below.
1. Place the cursor at the end of the last ```#region``` in ```Class1``` and hit {Enter}

**RESULTS**:
VS becomes unresponsive and newline is not inserted for 5-6 seconds...

```
#define Var1

using System;
using System.Collections.Generic;

class Class1
{
#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif

#if (false && Var1)
    Dictionary<int, List<string>> x = new Dictionary<int, List<string>>() {
    #region Region 1
        {
            1,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 2
        {
            2,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 3
        {
            3,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 4
        {
            4,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 5
        {
            5,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 6
        {
            6,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 7
        {
            7,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 8
        {
           8,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 9
        {
            9,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        },
    #endregion
    #region Region 10
        {
            10,
            new List<string>()
            {
                "a",
                "b",
                "c",
                "d",
                "e",
                "f"
            }
        }
    #endregion
    };
#endif
}

class Program
{
    static void Main()
    {
    }
}
```

### PR DESCRIPTION
**Pull request by rchande**
Avoid repeatedly calling ISyntaxFactsService.IsInInactiveRegion, which calls SyntaxTree.GetToken. GetToken needs to traverse directive trivia to find the token it is
attached to. This results in n ^ 2 traversals of the tree if there are repeated directive before the line to indent. Additionally, when the indenter determines
that it's examining a line inside a disabled region, instead of examining the previous line, it examines the first line preceeding the current line that isn't
inside a disabled region.

Fixes #5394.

### CODE REVIEW

**Comment by rchande** [comment]
@pharring @heejaechang Can you take a look?

**Comment by rchande** [comment]
retest this please

**Comment by rchande** [comment]
With this change (built release, on my machine), typing enter to get smart indentation is as responsive as any other character.

File: src/EditorFeatures/CSharpTest/Formatting/Indentation/SmartIndenterTests_Performance.cs [hunk 1]
```diff
@@ -0,0 +1,2217 @@
+﻿using System;
```
Comment (pharring) [inline src/EditorFeatures/CSharpTest/Formatting/Indentation/SmartIndenterTests_Performance.cs]: Needs OSS header

File: src/EditorFeatures/Core/Implementation/SmartIndent/AbstractIndentationService.AbstractIndenter.cs [hunk 2]
```diff
@@ -114,43 +117,56 @@ protected ITextSnapshotLine GetPreviousNonBlankOrPreprocessorLine()
...
+                    if (string.IsNullOrWhiteSpace(actualLine.GetText()))
                     {
-                        return false;
+                        lineNumber--;
+                        continue;
                     }
 
-                    // okay, now check whether it is preprocessor line or not
+                    // No preprocessors in the entire tree, so this line
+                    // line definitely doesn't have one
```
Comment (pharring) [inline src/EditorFeatures/Core/Implementation/SmartIndent/AbstractIndentationService.AbstractIndenter.cs]: repeated word "line"

**Comment by pharring** [comment]
Looks good. Please also re-test the scenario that led to #3515 (I can help you find the internal bug number, but it's also in the mail thread from earlier today)

**Comment by rchande** [comment]
@pharring Retested that scenario; looks good.

**Comment by KevinH-MS** [comment]
Played around a bit with updated bits...  LGTM :+1:

**Pull request merged**